fix: infer default namespace unless provided by provider config

rawkode opened the pull request:
### Proposed changes

Use the discovered kubeconfig namespace when one isn't provided by the Provider configuration instead of default.

### Related issues (optional)

Fix #971

viveklak commented on provider/pkg/provider/provider.go:
```diff
@@ -683,10 +683,11 @@ func (k *kubeProvider) Configure(_ context.Context, req *pulumirpc.ConfigureRequ
-	namespace := "default"
-	if k.defaultNamespace != "" {
+	namespace, _, err := kubeconfig.Namespace()
+	if err != nil || k.defaultNamespace != "" {
```
I think you want to set defaultNamespace to kubeconfig.Namespace() if it is non-empty and if the default namespace is not explicitly specified for the provider.

rawkode replied:
Maybe I misunderstood, but `k.defaultNamespace` is the namespace configured in the `k8s.Provider`.

The original code sets `namespace` to that value, so I thought it best to do the same; I don't believe `k.defaultNamespace` is used any further?

rawkode replied:
Though `k.defaultNamespace` is also used at line 1279. I'm not terribly familiar with this code.

rawkode commented:
@viveklak OK, I spent some time this morning going through this code properly and I understand the problem now. I've updated the PR.

I have no idea how to write a unit test for the `Configure` method or if that's even possible, so I've done manual testing locally with the built provider binary.

Setup:

`pulumi new kubernetes-typescript`

Test 1:

- I added `namespace: test-me` to the context for my local Kubernetes cluster
- `pulumi up`
- Resource was created in `test-me` namespace

Test 2:

- Continue from above
- `pulumi config set kubernetes:namespace default`
- `pulumi up`
- Resource is replaced and deployed to `default` namespace

I believe this is the behaviour that we need to resolve #971 

cc @lblackstone

lblackstone commented on CHANGELOG.md:
```diff
@@ -1,3 +1,7 @@
+## Unreleased
+
+- Infer default namespace from kubeconfig when not configured via the provider
```
```suggestion
- Infer default namespace from kubeconfig when not configured via the provider (https://github.com/pulumi/pulumi-kubernetes/pull/1896)
```

lblackstone commented:
/run-acceptance-tests
